consumers/sender/main.go

```diff
@@ -0,0 +1,232 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ses"
+	"github.com/nsqio/go-nsq"
+	"github.com/sirupsen/logrus"
+
+	"github.com/mailbadger/app/consumers"
+	"github.com/mailbadger/app/emails"
+	"github.com/mailbadger/app/entities"
+	"github.com/mailbadger/app/storage"
+	"github.com/mailbadger/app/utils"
+)
+
+const (
+	CharSet = "UTF-8"
+)
+
+// MessageHandler implements the nsq handler interface.
+type MessageHandler struct {
+	s storage.Storage
+}
+
+// HandleMessage is the only requirement needed to fulfill the
+// nsq.Handler interface.
+func (h *MessageHandler) HandleMessage(m *nsq.Message) error {
+	if len(m.Body) == 0 {
+		logrus.Error("Empty message, unable to proceed.")
+		return nil
+	}
+
+	msg := new(entities.SenderTopicParams)
+	err := json.Unmarshal(m.Body, msg)
+	if err != nil {
+		logrus.WithField("body", string(m.Body)).
+			WithError(err).Error("Malformed JSON message.")
+		return nil
+	}
+
+	if msg.SesKeys == nil {
+		logrus.WithFields(logrus.Fields{
+			"uuid":          msg.UUID,
+			"user_id":       msg.UserID,
+			"campaign_id":   msg.CampaignID,
+			"subscriber_id": msg.SubscriberID,
+		}).Error("SES Keys are nil.")
+		return nil
+	}
+
+	client, err := emails.NewSesSender(msg.SesKeys.AccessKey, msg.SesKeys.SecretKey, msg.SesKeys.Region)
+	if err != nil {
+		logrus.WithFields(logrus.Fields{
+			"uuid":          msg.UUID,
+			"user_id":       msg.UserID,
+			"campaign_id":   msg.CampaignID,
+			"subscriber_id": msg.SubscriberID,
+		}).WithError(err).Error("Unable to create SES sender")
+		return nil
+	}
+
+	_, err = h.s.GetSendLogByUUID(msg.UUID)
```

Comment:
Maybe we should check the send log status, if it's failed and retryable we can retry it again?

Reply:
I think we should talk about that in details